include/dali/c_api.h

```diff
@@ -139,13 +139,14 @@ DLL_PUBLIC void daliDeserializeDefault(daliPipelineHandle *pipe_handle,
  *                   Can be set to NULL.
  * @param stream CUDA stream to use when copying the data onto GPU. Remember to synchronize on the
  *               provided stream.
+ * @param sync If block until data provided is copied to the internal DALI buffer
  */
 DLL_PUBLIC void
 daliSetExternalInputAsync(daliPipelineHandle *pipe_handle, const char *name,
                           device_type_t device, const void *data_ptr,
                           dali_data_type_t data_type, const int64_t *shapes,
                           int sample_dim, const char *layout_str,
-                          cudaStream_t stream);
+                          cudaStream_t stream, int sync);
```

Comment:
There is a question if we still need both async and sync variants of this function if this can be handled by the parameter.

Reply:
Sync has own stream, in this variant you still need to provide one.

Reply:
Ok, makes sense.